WordPress/Classes/Utility/ImmuTable.swift

```diff
@@ -0,0 +1,206 @@
+    }
+
+    public static func registerRows(rows: [ImmuTableRow.Type], tableView: CellRegistrator) {
+        let registrables = rows.reduce([:]) {
+            (var classes, row) -> [String: CellRegistrable] in
+
+            classes[row.reusableIdentifier] = row.registrable
+            return classes
+        }
+        for (identifier, registrable) in registrables {
+            tableView.register(registrable, cellReuseIdentifier: identifier)
+        }
+    }
+}
+
+public class ImmuTableDataSource: NSObject, UITableViewDataSource {
+    var viewModel: ImmuTable
+    var configureCell: ((UITableViewCell) -> Void)?
+
+    init(viewModel: ImmuTable) {
+        self.viewModel = viewModel
+    }
+
+    public func numberOfSectionsInTableView(tableView: UITableView) -> Int {
+        return viewModel.sections.count
+    }
+
+    public func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return viewModel.sections[section].rows.count
+    }
+
+    public func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        let row = viewModel.rowAtIndexPath(indexPath)
+        let cell = tableView.dequeueReusableCellWithIdentifier(row.reusableIdentifier, forIndexPath: indexPath)
+
+        row.configureCell(cell)
+
+        configureCell?(cell)
+
+        return cell
+    }
+}
+
+public class ImmuTableDelegate: NSObject, UITableViewDelegate {
+    var viewModel: ImmuTable
+
+    init(viewModel: ImmuTable) {
+        self.viewModel = viewModel
+    }
+
+    public func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
+        let row = viewModel.rowAtIndexPath(indexPath)
+        if let action = row.action {
+            action(row)
+        }
+    }
+
+    public func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
+        let row = viewModel.rowAtIndexPath(indexPath)
+        if let customHeight = row.dynamicType.customHeight {
+            return CGFloat(customHeight)
+        }
+        return tableView.rowHeight
+    }
+}
+
+public struct ImmuTableViewHandler {
+    let target: UITableViewController
+
+    init(takeOver target: UITableViewController) {
+        self.target = target
+        self.target.tableView.dataSource = dataSource
+        self.target.tableView.delegate = delegate
+    }
+
+    var viewModel = ImmuTable(sections: []) {
+        didSet {
+            dataSource.viewModel = viewModel
+            delegate.viewModel = viewModel
+            if target.isViewLoaded() {
+                target.tableView.reloadData()
+            }
+        }
+    }
+
+    lazy var dataSource: ImmuTableDataSource = {
+        return ImmuTableDataSource(viewModel: self.viewModel)
+    }()
+    
+    lazy var delegate: ImmuTableDelegate = {
+        return ImmuTableDelegate(viewModel: self.viewModel)
+    }()
+}
+
+protocol CustomImmuTableRow: ImmuTableRow {
```

Comment:
>  just set as optional whatever is not mandatory

I tried using optional protocol methods (not sure if here or somewhere else), until I ran into this little issue:

> Optional protocol requirements can only be specified if your protocol is marked with the @objc attribute
> [...]
> Note also that @objc protocols can be adopted only by classes that inherit from Objective-C classes or other @objc classes. They can’t be adopted by structures or enumerations